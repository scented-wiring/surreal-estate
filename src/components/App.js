import React, { useState } from "react";
import "../styles/App.css";
import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";
import { Switch, Route } from "react-router-dom";

const App = () => {
  const [userID, setUserID] = useState("");

  const handleLogin = (response) => {
    setUserID(response.userID);
  };

  const handleLogout = () => window.FB.logout(() => setUserID(""));

  return (
    <div className="App">
      <NavBar onLogin={handleLogin} onLogout={handleLogout} userID={userID} />
      <Switch>
        <Route
          exact
          path="/"
          render={(props) => <Properties {...props} userID={userID} />}
        />
        <Route exact path="/add-property" component={AddProperty} />
      </Switch>
    </div>
  );
};

export default App;
