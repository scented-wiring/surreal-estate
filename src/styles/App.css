@import url("https://fonts.googleapis.com/css2?family=Noto+Sans+JP:wght@500&display=swap");

html,
body {
  margin: 0;
  background-color: white;
}

.app {
  display: flex;
  flex-direction: column;
  font-family: "Noto Sans JP", sans-serif;
  font-size: 1em;
}

.component {
  flex: 1;
}

a {
  color: black;
}

a:link {
  text-decoration: none;
}

a:visited {
  text-decoration: none;
}

a:hover {
  text-decoration: none;
  color: #ad8a67;
}

a:active {
  text-decoration: none;
}

button {
  border: none;
  background-color: #402e32;
}

button:hover {
  color: #ffad48;
  transition: all 0.4s ease 0s;
}

@media only screen and (max-width: 974px) {
  .app {
    font-size: 0.9em;
  }
}

@media (max-width: 767px) {
  .large-screen {
    display: none;
  }
}
