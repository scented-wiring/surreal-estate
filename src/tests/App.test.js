import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import App from "../components/App";

beforeAll(() => {
  const script = document.createElement("script");
  document.body.appendChild(script);
});

describe("App", () => {
  it("renders a component", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    expect(asFragment).toMatchSnapshot();
  });

  it("renders a Navbar", () => {
    const { getByText } = render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    const title = getByText("Surreal Estate");
    expect(title).toBeInTheDocument();
  });
});
