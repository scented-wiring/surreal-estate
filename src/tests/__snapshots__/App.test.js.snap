// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders a component 1`] = `[Function]`;

exports[`App renders correctly 1`] = `[Function]`;
